### PR TITLE
Document the allow_user_specified_id settings

### PR DESCRIPTION
**Pull request by fmoor**

### CODE REVIEW

**Comment by fmoor** [comment]
Is there more that should be said about this setting?

File: docs/stdlib/cfg.rst [hunk 1]
```diff
@@ -85,6 +85,9 @@ Query planning
 Client connections
 ------------------
 
+:eql:synopsis:`allow_user_specified_id -> bool`
+  Makes it possible to set the `.id` property when inserting new objects.
```
Comment (elprans) [inline docs/stdlib/cfg.rst:89]: We should add a warning about the security implications of this and that custom object ids should come from a trusted source.

File: docs/stdlib/cfg.rst [hunk 2]
```diff
@@ -85,6 +85,19 @@ Query planning
...
+
+  Enabling this feature introduces some security vulnerabilities:
+
+  1. An unprivileged user can discover ids that already exist in the database
+     by trying to insert new values and noting when there is a constraint
+     violation on `.id` even if the user doesn't have access to the relevant
```
Comment (fmoor) [inline docs/stdlib/cfg.rst:95]: I'm not sure I got all the details right. I pretty much just copied this: https://github.com/edgedb/edgedb/issues/3510#issuecomment-1189419879

File: docs/stdlib/cfg.rst (hunk 1 above)
Comment (fmoor) [reply docs/stdlib/cfg.rst:89]: added

**Review by 1st1 — APPROVED** [review]

**Pull request merged**